traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -1854,41 +1838,134 @@ func createV4(inf *api.APIInfo, w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	resultRows, err := inf.Tx.NamedQuery(insertQueryV4, server)
+	serverID, err := createServerV4(inf.Tx, server)
 	if err != nil {
 		userErr, sysErr, errCode := api.ParseDBError(err)
-		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
 		return
 	}
-	defer resultRows.Close()
+
+	userErr, sysErr, errCode := createInterfaces(int(serverID), server.Interfaces, inf.Tx.Tx)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+
+	if server.UpdPending != nil && *server.UpdPending { // To continue to work with the legacy implementation and priority. However, consider bool UpdPending deprecated
+		if err := dbhelpers.QueueUpdateForServer(inf.Tx.Tx, serverID); err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+			return
+		}
+	} else if server.ConfigUpdateTime != nil {
+		if err := dbhelpers.QueueUpdateForServerWithTime(inf.Tx.Tx, serverID, *server.ConfigUpdateTime); err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+			return
+		}
+	}
+
+	if server.RevalPending != nil && *server.RevalPending { // To continue to work with the legacy implementation and priority. However, consider bool RevalPending deprecated
+		if err := dbhelpers.QueueRevalForServer(inf.Tx.Tx, serverID); err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+			return
+		}
+	} else if server.RevalUpdateTime != nil {
+		if err := dbhelpers.QueueRevalForServerWithTime(inf.Tx.Tx, serverID, *server.RevalUpdateTime); err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+			return
+		}
+	}
```

Comment:
ditto an earlier comment: I don't think the "create/update server" APIs should deal with queuing/unqueuing updates/revals at all. There is another API for doing that already that should be the sole owner of this functionality IMO

Reply:
Ditto as well. I put various replies about this in other comments.

Should we remove the ability to set any config/reval updates via these server endpoints (PUT/POST/PATCH)?

Reply:
Removed